components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/FaultGatewaysException.java

```diff
@@ -0,0 +1,58 @@
+/*
+*  Copyright (c) 2005-2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.carbon.apimgt.api;
+
+import java.util.List;
+import java.util.Map;
+
+public class FaultGatewaysException extends Exception {
+	private Map<String, List<String>> faultMap;
+
+	public String getFaultGateWayString() {
+		String failedJson = "{\"PUBLISHED\" : \"\" ,\"UNPUBLISHED\":\"\"}";
+		if (faultMap != null) {
+			if (!faultMap.isEmpty()) {
+				StringBuilder failedToPublish = new StringBuilder();
+				StringBuilder failedToUnPublish = new StringBuilder();
+				for (String environmentName : faultMap.get("PUBLISHED")) {
+					failedToPublish.append(environmentName + ",");
+				}
+				for (String environmentName : faultMap.get("UNPUBLISHED")) {
+					failedToUnPublish.append(environmentName + ",");
+				}
+				if (!"".equals(failedToPublish.toString())) {
```

Comment:
Same behaviour could be achieved by checking length of the StringBuilder object.